scylla/src/transport/connection.rs

```diff
@@ -2023,19 +2023,15 @@ mod tests {
                     let futs = (base..base + batch_size).map(|j| {
                         let q = Query::new("INSERT INTO t (p, v) VALUES (?, ?)");
                         let conn = conn.clone();
-                        async move {
-                            conn.query(&q, (j, vec![j as u8; j as usize]), None)
-                                .await
-                                .unwrap()
-                        }
+                        async move { conn.query(&q, (j, vec![j as u8; j as usize]), None).await }
                     });
-                    futures::future::join_all(futs).await;
+                    futures::future::try_join_all(futs).await
                 }));
 
                 tokio::task::yield_now().await;
             }
 
-            futures::future::join_all(futs).await;
+            futures::future::try_join_all(futs).await.unwrap();
```

Comment:
On the other hand, I'm not sure this line is correct now after your changes. If you look at the definition of `tokio::task::spawn`, you will see that it returns a handle that is a future which returns its own `Result` - if the future polled by the tokio task panics, the handle will contain `Err`. So, because you changed the futures to return `Result<>`s, now `try_join_all` returns a `Result<Result<_>>`.

Assuming that you won't change the `futs.push` part of the code, you will need to do two unwraps here. On the other hand, if you revert changes you did to the `futs.push` part, you should keep the line directly above as it is and that's it.

To sum up, the whole change should be an one-liner.

Reply:
Ahh you're right. It's a shame that rust didn't complain about unused Results here.
I uploaded new version which also checks that the result of `join_all` is always a `Vec<()>`